api/src/pcapi/core/offers/api.py

```diff
@@ -894,21 +894,18 @@ def add_criteria_to_offers(
 
 
 def reject_inappropriate_products(ean: str, send_booking_cancellation_emails: bool = True) -> bool:
-    products = models.Product.query.filter(models.Product.extraData["ean"].astext == ean).all()
+    product_query = models.Product.query.filter(models.Product.extraData["ean"].astext == ean)
+    product_ids = [p.id for p in product_query.with_entities(models.Product.id).all()]
```

Comment:
(détail) l'appel à `.all()` n'est pas nécessaire.

Reply:
@Meewan semblait dire que c'était nécessaire donc j'ai un doute.

Reply:
Ça n'est pas nécessaire (mais ça n'est pas très important).

Reply:
J'ignorais qu'on pouvais itérer sur un obet query.